main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -561,7 +562,139 @@ object Lowering {
       val t = visitType(tpe)
       Expression.Spawn(e, t, pur, eff, loc)
 
-    case Expression.Par(exp, _) =>
+    case Expression.Par(Expression.Apply(exp, exps, tpe, pur, eff, loc1), loc0) => {
+      /*
+      {
+        let ch1 = channel 1;
+        let ch2 = channel 1;
+        let ch3 = channel 1;
+        ...
+        let chn = channel 1;
+
+        spawn ch1 <- exp1;
+        spawn ch2 <- exp2;
+        spawn ch3 <- exp3;
+        ...
+        spawn chn <- expn;
+
+        let tmp1 = <- ch1;
+        let tmp2 = <- ch2;
+        let tmp3 = <- ch3;
+        ...
+        let tmpn = <- chn;
+
+        (tmp1)(tmp2, tmp3, ..., tmpn) // these last two is a Let(tpmn = <- chn, apply) expression.
+      } as & *the effect of tmp1 to tmpn*
+       */
+
+
+      def mkChannel(e: Expression): Expression = {
+        val loc = e.loc.asSynthetic
+        Expression.NewChannel(Expression.Int32(1, loc), Type.mkChannel(e.tpe, loc), Type.Impure, Type.Empty, loc) // Should have IO effect?
+      }
+
+      def mkPutChannel(chan: Symbol.VarSym, e: Expression): Expression = {
+        val loc = e.loc.asSynthetic
+        Expression.PutChannel(Expression.Var(chan, Type.mkChannel(e.tpe, loc), loc), e, e.tpe, Type.Impure, e.eff, loc)
+      }
+
+      def mkSpawn(chan: Symbol.VarSym, e: Expression): Expression = {
+        Expression.Spawn(mkPutChannel(chan, e), Type.Unit, e.pur, e.eff, e.loc.asSynthetic)
+      }
+
+      def mkVarExp(sym: Symbol.VarSym, tpe: Type, loc: SourceLocation): Expression = {
+        Expression.Var(sym, tpe, loc)
+      }
+
+      def mkWait(chan: Expression): Expression = {
+        Expression.GetChannel(chan, chan.tpe, Type.Impure, chan.eff, chan.loc.asSynthetic)
```

Comment:
Chan.tpe is incorrect. Because that would be `Channel[String]` and not `String` as you would expect.